### PR TITLE
add autogen table/column comment

### PR DESCRIPTION
**Pull request by sfc-gh-rehuang**
Use cortex LLM function to auto-generate comments for table and columns

### CODE REVIEW

File: semantic_model_generator/snowflake_utils/snowflake_connector.py [hunk 1]
```diff
@@ -83,18 +138,29 @@ def get_table_representation(
     columns_df: pd.DataFrame,
     max_workers: int,
 ) -> Table:
-    table_comment = columns_df[_TABLE_COMMENT_COL].iloc[0]
+    table_comment, is_auto_generated = _get_table_comment(conn, table_name, columns_df)
+
+    def _get_ndv_per_column(column_row: pd.Series, ndv_per_column: int) -> int:
```
Comment (sfc-gh-cnivera) [inline semantic_model_generator/snowflake_utils/snowflake_connector.py:143]: dumb q - what's `ndv`? number of distinct values?

**Review by sfc-gh-cnivera — APPROVED** [review]

File: semantic_model_generator/snowflake_utils/snowflake_connector.py (hunk 1 above)
Comment (sfc-gh-rehuang) [reply semantic_model_generator/snowflake_utils/snowflake_connector.py:138]: yes, number of distinct values!

File: semantic_model_generator/snowflake_utils/snowflake_connector.py [hunk 2]
```diff
@@ -74,6 +78,49 @@
...
+                .execute(f"select get_ddl('table', '{table_name}');")
+                .fetchall()[0][0]
+            )
+            comment_prompt = f"Here is a table with below DDL: {tbl_ddl} \nPlease provide a business description for the table. Only return the description without any other text."
+            complete_sql = f"select SNOWFLAKE.CORTEX.COMPLETE('{_autogen_model}', '{comment_prompt}')"
+            cmt = conn.cursor().execute(complete_sql).fetchall()[0][0]  # type: ignore[union-attr]
+            return str(cmt + AUTOGEN_TOKEN)
+        except Exception as e:
+            logger.warning(f"Unable to auto generate table comment: {e}")
+            return ""
```
Comment (sfc-gh-dasilva) [inline semantic_model_generator/snowflake_utils/snowflake_connector.py:100]: should we make this a flag that the user controls?

**Review by sfc-gh-dasilva — APPROVED** [review]

File: semantic_model_generator/snowflake_utils/snowflake_connector.py (hunk 2 above)
Comment (sfc-gh-rehuang) [reply semantic_model_generator/snowflake_utils/snowflake_connector.py:100]: Thought about it, but also don't see much cases user would not want to generate

**Pull request merged**